src/_micro-api/libraries/signalDecoder/src/signalDecoder.cpp

```diff
@@ -43,6 +43,40 @@
 #endif
 
 
+
+/* **********************************************************
+ * SELFMADE - implementation of non standard function itoa()
+ * "itoa() function is not defined in ANSI-C and is not part of C++, but is supported by some compilers."
+ */
+
+char* myitoa(int num, char *str) {
+  static const uint16_t pows10[] = {10000, 1000, 100, 10};
```

Comment:
Gibt es einen speziellen Grund, weshalb Du die Variable static gewählt hast?

Reply:
Es wird mit Sicherheit aus der Vorlage stammen. Richtig @elektron-bbs ?

@sidey79 , wir haben ja mindestens 5 Varianten verglichen und dann diese gewählt welche am kleinsten und schnellsten arbeitete.

Reply:
static sorgt ja dafür, dass dieser array beim verlassen der funktion weiterhin im stack bleibt und der Speicherplatz nicht freigegeben wird.

Reply:
Ich habe die Variante ohne "static" mal in meinen Testsketch aufgenommen, mit dem Ergebnis, das der Sketch dadurch 56 Byte größer wird und die Ausführung 36 µS länger dauert. Leider verweigert mir Github jetzt das Hochladen des Sketches (weder ZIP noch TXT). Deshalb hier nur die Übersicht (Var 4 und 41 sind die beiden zu vergleichenden):

```
/* Board: Arduino Pro Micro = Arduino Leonardo (ATMega32u4)

  Var   Sketch        Global   micros   result             use
  ----------------------------------------------------------------------------------------------------------------------
  0     5668 (base)   229       64      tut nix            nix
  1     5866 (+198)   229      544      OK                 stdlib itoa()
  2     5954 (+286)   229      680      ERROR -32768       SELFMADE - implementation of non standard function itoa()
  21    5922 (+254)   229      656      ERROR -32768       optimzed SELFMADE - implementation of non standard function itoa()
  3                                                        not tested from https://www.mikrocontroller.net/topic/413972#4820985
  4     5796 (+128)   237      336      OK                 modified from https://www.mikrocontroller.net/topic/413972#4828913
  41    5852 (+184)   237      372      OK                 without static modified from https://www.mikrocontroller.net/topic/413972#4828913
  5     5842 (+174)   229      548      OK                 https://stackoverflow.com/questions/21501815/optimal-base-10-only-itoa-function
  6     5966 (+298)   301      680      OK                 Written by Lukás Chmela
  61    5926 (+258)   249      692      OK                 Written by Lukás Chmela, modified for only base 10
*/
```

Reply:
Verblüffend. Ob dafür der präprozessor verantwortlich ist?
Ich war der Meinung const Variablen gehen immer in den Flash.

pows10[] ist ein Array mit unbekannter Größe.
Vermutlich findet auch hier der präprozessor heraus, dass es vier Werte sind aber es könnte nicht schaffen die Arraygrößen fix zu definieren oder?

Reply:
> Verblüffend. Ob dafür der präprozessor verantwortlich ist?
> Ich war der Meinung const Variablen gehen immer in den Flash.

Womit das genau zusammen hängt, weiß ich nicht.

> pows10[] ist ein Array mit unbekannter Größe.
> Vermutlich findet auch hier der präprozessor heraus, dass es vier Werte sind aber es könnte nicht schaffen die Arraygrößen fix zu definieren oder?

Schaden kann es sicher nichts, pows10[] durch pows10[4] zu ersetzen. Ersichtliche Vorteile bringt es aber auch nicht.